### PR TITLE
space-station-14-launcher: init at 0.20.5

### PR DESCRIPTION
**Pull request by gradientvera**
###### Description of changes

[Launcher](https://github.com/space-wizards/SS14.Launcher) for [Space Station 14](https://spacestation14.io/), a Free and Open Source multiplayer roleplaying game.
This is my first time writing a Nix package, so I'd appreciate any feedback.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16999,6 +16999,11 @@
     githubId = 393108;
     name = "Damien Diederen";
   };
+  zumorica = {
+    name = "Vera Aguilera Puerto";
+    github = "Zumorica";
+    githubId = 6766154;
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:17006]: You should put that into it's own commit

File: pkgs/games/space-station-14-launcher/default.nix [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    preFixup = ''
+      makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    '';
+
+    meta = with lib; {
+      description = "Launcher for Space Station 14, a multiplayer game about paranoia and disaster";
+      homepage = "https://spacestation14.io";
+      license = licenses.mit;
+      maintainers = with maintainers; [ zumorica ];
```
Comment (Janik-Haag) [inline pkgs/games/space-station-14-launcher/default.nix:108]:
```suggestion
      maintainers = [ maintainers.zumorica ];
```

**Comment by Janik-Haag** [comment]
Since it its the init of the module and still a open PR you should merge the formating and space-station-14 init commits. You can do that with something like that.

The hash in git commit is the one of the commit you want ot amend to and the one in git rebase is the one before you changed anything in the git history.
I just put in example values ^^
```
git commit --fixup amend:06b0cf0241f 
git rebase -i --autosquash 33c9fd85e52
```

**Review by Janik-Haag — APPROVED** [review]

File: pkgs/games/space-station-14-launcher/default.nix [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+, wrapGAppsHook
+, buildFHSUserEnv
+, runCommand
+, iconConvTools
+, copyDesktopItems
+, makeDesktopItem
+}:
+
+let
+  rev = "0.20.5";
```
Comment (SuperSandro2000) [inline pkgs/games/space-station-14-launcher/default.nix:38]:
```suggestion
  version = "0.20.5";
```

File: pkgs/games/space-station-14-launcher/default.nix [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  space-station-14-launcher = buildDotnetModule rec {
+    inherit pname;
+
+    # Workaround to prevent buildDotnetModule from overriding assembly versions.
+    name = "${pname}-${rev}";
+
+    src = fetchFromGitHub {
+      owner = "space-wizards";
+      repo = "SS14.Launcher";
+      rev = "v${rev}";
```
Comment (SuperSandro2000) [inline pkgs/games/space-station-14-launcher/default.nix:49]:
```suggestion
      rev = "v${version}";
```

File: pkgs/games/space-station-14-launcher/default.nix [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+    buildType = "Release";
+    selfContainedBuild = false;
+
+    projectFile = [
+      "SS14.Loader/SS14.Loader.csproj"
+      "SS14.Launcher/SS14.Launcher.csproj"
+    ];
+
+    nugetDeps = ./deps.nix; # File generated with `nix-build -A space-station-14-launcher.passthru.fetch-deps`
```
Comment (SuperSandro2000) [inline pkgs/games/space-station-14-launcher/default.nix:62]: This should ideally be put into an update script

File: pkgs/games/space-station-14-launcher/default.nix [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+  # Workaround for hardcoded soundfont paths in downloaded engine assemblies.
+  soundfont-fluid-fixed = runCommand "soundfont-fluid-fixed" { } ''
+    mkdir -p "$out/share/soundfonts"
+    ln -sf ${soundfont-fluid}/share/soundfonts/FluidR3_GM2-2.sf2 $out/share/soundfonts/FluidR3_GM.sf2
+  '';
+
+in
+buildFHSUserEnv rec {
+  name = "${pname}-${rev}";
```
Comment (SuperSandro2000) [inline pkgs/games/space-station-14-launcher/default.nix:151]:
```suggestion
  name = "${pname}-${version}";
```

File: pkgs/games/space-station-14-launcher/default.nix [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        categories = [ "Game" ];
+        startupWMClass = meta.mainProgram;
+      })
+    ];
+
+    postInstall = ''
+      mkdir -p $out/lib/${pname}/loader
+      cp -r SS14.Loader/bin/${buildType}/*/* $out/lib/${pname}/loader/
+
+      icoFileToHiColorTheme SS14.Launcher/Assets/icon.ico ${pname} $out
```
Comment (SuperSandro2000) [inline pkgs/games/space-station-14-launcher/default.nix:124]: Usually we do not replace pname wherever possible to reduce code complexity

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -16999,6 +16999,11 @@
     githubId = 393108;
     name = "Damien Diederen";
   };
+  zumorica = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:17002]: It would be nice if you add another mean of contact like email or matrix.

File: pkgs/games/space-station-14-launcher/default.nix (hunk 5 above)
Comment (Zumorica) [reply pkgs/games/space-station-14-launcher/default.nix]: I made an update script. However to make it work properly I had to made a separate file for the unwrapped launcher.

**Comment by ashkitten** [comment]
works great, lgtm!

File: pkgs/games/space-station-14-launcher/space-station-14-launcher.nix [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+, zlib
+, glib
+, gdk-pixbuf
+}:
+let
+  version = "0.20.5";
+  pname = "space-station-14-launcher";
+in
+buildDotnetModule rec {
+  inherit pname;
```
Comment (SuperSandro2000) [inline pkgs/games/space-station-14-launcher/space-station-14-launcher.nix:38]:
```suggestion
  inherit pname version;
```

Comment (Zumorica) [reply pkgs/games/space-station-14-launcher/space-station-14-launcher.nix:38]: Doing this would break the launcher, as `buildDotnetModule` will use this to version the built SS14.Loader assembly (which shouldn't be versioned) and then the downloaded engine assemblies won't be able to load it properly.
